pkg/services/query/query_test.go

```diff
@@ -319,6 +319,57 @@ func TestQueryDataMultipleSources(t *testing.T) {
 		assert.Len(t, header.Values("test"), 2)
 	})
 
+	t.Run("forward headers are sent to the datasource", func(t *testing.T) {
+		tc := setup(t)
+		query1, err := simplejson.NewJson([]byte(`
+			{
+				"datasource": {
+					"type": "mysql",
+					"uid": "ds1"
+				},
+				"refId": "A"
+			}
+		`))
+		require.NoError(t, err)
+		query2, err := simplejson.NewJson([]byte(`
+			{
+				"datasource": {
+					"type": "mysql",
+					"uid": "ds2"
+				},
+				"refId": "B"
+			}
+		`))
+		require.NoError(t, err)
+		queries := []*simplejson.Json{query1, query2}
+		reqDTO := dtos.MetricRequest{
+			From:    "2022-01-01",
+			To:      "2022-01-02",
+			Queries: queries,
+			Debug:   false,
+		}
+
+		req, err := http.NewRequest("POST", "http://localhost:3000", nil)
+		require.NoError(t, err)
+		reqCtx := &contextmodel.ReqContext{
+			SkipQueryCache: false,
+			Context: &web.Context{
+				Resp: web.NewResponseWriter(http.MethodGet, httptest.NewRecorder()),
+				Req:  req,
+			},
+		}
+		ctx := ctxkey.Set(context.Background(), reqCtx)
+
+		_, err = tc.queryService.QueryData(ctx, tc.signedInUser, true, reqDTO, QueryDataOptionWithForwardHeaders(map[string]string{"X-forward-header": "some-val"}))
+		require.NoError(t, err)
+
+		// response headers should be merged
+		header := contexthandler.FromContext(ctx).Resp.Header()
+		assert.Len(t, header.Values("test"), 2)
```

Comment:
where was this header added and why do we want to assert this?

Reply:
Discussed offline: This assert extends from the above test which checks for multiple datasources. Moving the test to a separate function.